Products/ZenModel/Device.py

```diff
@@ -2235,4 +2236,22 @@ def getMacAddressCache(self):
     def getMacAddresses(self):
         return list(self.macaddresses or [])
 
+    security.declareProtected(access_contents_information, 'propertyItems')
+    def propertyItems(self):
+        """Return a list of (id, property) tuples. If property id related
+           to password, it will mask property value with '***'
+        """
+        result = []
+        for prop in self._properties:
+            prop_id = prop['id']
+            if prop['type'] == 'password':
+                prop_value = "***"
```

Comment:
well, unfortunately I have a lack of ideas to fix it. The only suggestion is to add an optional parameter to the `propertyItems()` function to pass it, if we need password in function output and mask password otherwise.